src/stack.rs

```diff
@@ -671,16 +703,15 @@ impl Cache {
         // Either way, start by populating a temporary file.
         let mut tmp = NamedTempFile::new_in(cache.temp_dir(key)?)?;
         populate(tmp.as_file_mut(), old)?;
-        fix_tempfile_permissions(&tmp)?;
-        self.maybe_sync(tmp.as_file())?;
 
         // Grab a read-only return value before publishing the file.
-        let path = tmp.path();
-        let mut ret = File::open(path)?;
+        let mut ret = File::open(tmp.path())?;
```

Comment:
the only thing I noticed while I was looking at `strace` output was this file is opened before finalize(), which means the contents may not be synced. However ret is not read from so I don't think it matters?

Reply:
yeah, it shouldn't matter but you're right that we might as well open after fsync.